### ISSUE TITLE
tch 0.11.0 and HEAD don't compile on macOS

### ISSUE THREAD

**Issue by GaryBoone** [LaurentMazare/tch-rs#689]
On macOS Ventura 13.3.1, x86 and M1, with the following minimal repro program:
```rust
use tch::{Device, Tensor};

fn main() {
    let t = Tensor::of_slice(&[1i64, 3]).to_device(Device::Cpu);
    println!("t =\n{}", t);
}
```
and 
```bash
$ export DYLD_LIBRARY_PATH=/opt/homebrew/Caskroom/miniforge/base/envs/rust_transformer/lib/python3.9/site-packages/torch/lib
$ export LIBTORCH=/opt/homebrew/Caskroom/miniforge/base/envs/rust_transformer/lib/python3.9/site-packages/torch/
```
Here are the results of compiling with three versions of tch:

* `tch = "0.10.3"`:
```shell
$ cargo run
...
t =
[1, 3]
Tensor[[2], Int64]
```

* `tch = { git = "https://github.com/LaurentMazare/tch-rs", default-features = false }`:
```shell
error[E0106]: missing lifetime specifier
  --> /Users/.../.cargo/git/checkouts/tch-rs-b9d1710db79374bb/abf8240/torch-sys/build.rs:35:62
   |
35 | fn get_pypi_wheel_url_for_aarch64_macosx() -> anyhow::Result<&str> {
   |                                                              ^ expected named lifetime parameter
   |
   = help: this function's return type contains a borrowed value, but there is no value for it to be borrowed from
help: consider using the `'static` lifetime
   |
35 | fn get_pypi_wheel_url_for_aarch64_macosx() -> anyhow::Result<&'static str> {
   |                                                               +++++++

error[E0308]: `if` and `else` have incompatible types
   --> /Users/.../.cargo/git/checkouts/tch-rs-b9d1710db79374bb/abf8240/torch-sys/build.rs:172:25
    |
162 | /                      if env::var("CARGO_CFG_TARGET_ARCH") == Ok(String::from("aarch64")) {
163 | |/                         get_pypi_wheel_url_for_aarch64_macosx().expect(
164 | ||                             "Failed to retrieve torch from pypi.  Pre-built version of libtorch for apple silicon a...
165 | ||                             You can install torch manually following the indications from https://github.com/Lauren...
166 | ||                             pip3 install torch=={TORCH_VERSION}
...   ||
169 | ||                             export DYLD_LIBRARY_PATH=${{LIBTORCH}}/lib
170 | ||                             ")
    | ||______________________________- expected because of this
171 | |                      } else {
172 | |                          format!("https://download.pytorch.org/libtorch/cpu/libtorch-macos-{TORCH_VERSION}.zip")
    | |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&str`, f
ound struct `String`
173 | |                      }
    | |______________________- `if` and `else` have incompatible types
    |
    = note: this error originates in the macro `format` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0106, E0308.
For more information about an error, try `rustc --explain E0106`.
error: could not compile `torch-sys` due to 2 previous errors
```

* `tch = "0.11.0"`:
```shell
warning: clang: warning: -Wl,-rpath=/opt/homebrew/Caskroom/miniforge/base/envs/rust_transformer/lib/python3.9/site-packages/tor
ch/lib: 'linker' input unused [-Wunused-command-line-argument]
warning: clang: warning: -Wl, rpath=/opt/homebrew/Caskroom/miniforge/base/envs/rust_transformer/lib/python3.9/site-packages/tor
ch/lib: 'linker' input unused [-Wunused-command-line-argument]
warning: In file included from libtch/torch_api.cpp:20:
warning: libtch/stb_image_write.h:742:13: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instea
d. [-Wdeprecated-declarations]
warning:       len = sprintf(buffer, "EXPOSURE=          1.0000000000000\n\n-Y %d +X %d\n", y, x);
warning:             ^
warning: /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
...
warning: 1 warning and 20 errors generated.

error: failed to run custom build command for `torch-sys v0.11.0`

Caused by:
  process didn't exit successfully: `/Users/.../Development/rust/ML/tch/tch_m1_minimal_test/target/debug/build/tor
ch-sys-898119953f453e8c/build-script-build` (exit status: 1)
...
```
... and more. 

Let me know if you would like these separated into two issues.

**Comment by LaurentMazare** [comment]
The `sprintf` warning has been fixed recently, #658. I just pushed #690 for the issue at the github tip, this was only happening when disabling default features hence wasn't covered by the CI (the CI tests the macos version but using the default features and downloading libtorch), so I would suggest trying again with the new github tip - sadly I don't have a mac to reproduce.

**Comment by LaurentMazare** [comment]
Closing this as hopefully the fix worked fine, feel free to re-open if not the case.
